nixos/modules/virtualisation/nixos-containers.nix

```diff
@@ -482,7 +482,8 @@ in
               '';
               type = lib.mkOptionType {
                 name = "Toplevel NixOS config";
-                merge = loc: defs: (import "${toString config.nixpkgs}/nixos/lib/eval-config.nix" {
+                # TODO (during review): clarify whether `lib` be extended with nixos-lib - remove this comment
+                merge = loc: defs: ((import "${toString config.nixpkgs}/nixos/lib" {}).evalSystemConfiguration {
```

Comment:
Should it? we already have `extendModules` in the `args`.

If it should, this is another PR, though.

Reply:
The point of a container is to have a strong barrier between it and the host. That includes evaluation afaic.
Conflating the Nixpkgs library and the `nixpkgs` flake `lib` was a huge mistake, but expensive to fix. :(

Reply:
removing that comment and (maybe) flagging for a follow up PR.